### PR TITLE
🏗 Typescriptify visual-diff task

### PR DESCRIPTION
**Pull request by danielrozenberg**
This doesn't actually work cause the task runner doesn't understand TypeScript files. I was wondering if you can help me with that bit?

### CODE REVIEW

File: build-system/tasks/visual-diff/verifiers.ts [hunk 1]
```diff
@@ -126,33 +122,32 @@ async function waitForPageLoad(
  *
  * Timeout is set to CSS_SELECTOR_RETRY_MS * CSS_SELECTOR_RETRY_ATTEMPTS ms.
  *
- * @param {!puppeteer.Page} page page to check the visibility of elements in.
- * @param {string} selector CSS selector for elements to wait on.
- * @param {!VisibilityDef} options with key 'visible' OR 'hidden' set to true.
- * @param {number} timeoutMillis how long to retry.
+ * @param page page to check the visibility of elements in.
```
Comment (rileyajones) [inline build-system/tasks/visual-diff/verifiers.ts:125]:
```suggestion
 * @param page to check the visibility of elements in.
```

**Review by rileyajones — APPROVED** [review]
LGTM on the typing, I'm love watching this transformation happen

**Comment by danielrozenberg** [comment]
> LGTM on the typing, I'm love watching this transformation happen

I think you missed the part where I said this isn't working and I need help 😅

**Comment by danielrozenberg** [comment]
Bump. `amp <task>` doesn't actually know how to load TS files, got any idea how to make progress here? :D

**Comment by samouri** [comment]
> Bump. `amp <task>` doesn't actually know how to load TS files, got any idea how to make progress here? :D

I personally don't think (but can be convinced!) that we should use `.ts` files for build-system code. For client side code we will always need a bundler -- and therefore using `.ts` incurs no additional cost.

For server-side code and CLIs, we get a performance boost by writing plain JS. We _are_ compatible with TS types in comments, so you get the benefits of the TS type system. You can also freely create `.d.ts` files and import elements within comments.

**Comment by danielrozenberg** [comment]
Closing because meh 😅

**Pull request closed without merging**